packages/twenty-server/src/modules/messaging/services/gmail-fetch-message-content-from-cache/gmail-fetch-message-content-from-cache-v2.service.ts

```
@@ -0,0 +1,303 @@
+import { MessageParticipantService } from 'src/modules/messaging/services/message-participant/message-participant.service';
+import { MessageService } from 'src/modules/messaging/services/message/message.service';
+import { ParticipantWithMessageId } from 'src/modules/messaging/types/gmail-message';
+import {
+  CreateCompanyAndContactJobData,
+  CreateCompanyAndContactJob,
+} from 'src/modules/connected-account/auto-companies-and-contacts-creation/jobs/create-company-and-contact.job';
+import { GmailFetchMessageContentFromCacheService } from 'src/modules/messaging/services/gmail-fetch-message-content-from-cache/gmail-fetch-message-content-from-cache.service';
+
+@Injectable()
+export class GmailFetchMessageContentFromCacheV2Service {
+  private readonly logger = new Logger(
+    GmailFetchMessageContentFromCacheService.name,
+  );
+
+  constructor(
+    private readonly fetchMessagesByBatchesService: FetchMessagesByBatchesService,
+    @InjectObjectMetadataRepository(ConnectedAccountWorkspaceEntity)
+    private readonly connectedAccountRepository: ConnectedAccountRepository,
+    @InjectObjectMetadataRepository(MessageChannelWorkspaceEntity)
+    private readonly messageChannelRepository: MessageChannelRepository,
+    @InjectCacheStorage(CacheStorageNamespace.Messaging)
+    private readonly cacheStorage: CacheStorageService,
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    @Inject(MessageQueue.messagingQueue)
+    private readonly messageQueueService: MessageQueueService,
+    private readonly messageService: MessageService,
+    private readonly messageParticipantService: MessageParticipantService,
+  ) {}
+
+  async fetchMessageContentFromCache(
+    workspaceId: string,
+    connectedAccountId: string,
+  ) {
+    const connectedAccount = await this.connectedAccountRepository.getById(
+      connectedAccountId,
+      workspaceId,
+    );
+
+    if (!connectedAccount) {
+      this.logger.error(
```

should throw and be in catch in job

---

should still throw when connectAccount, messageChannel are not found